src/mail/mail.service.ts

```diff
@@ -0,0 +1,100 @@
+import { Injectable } from "@nestjs/common";
+import axios, { AxiosResponse } from "axios";
+import * as ejs from "ejs";
+
+@Injectable()
+export class MailService {
+  private readonly mailgunAPIBaseURL = `${process.env.MAILGUN_API_URL}/${process.env.MAILGUN_EMAIL_DOMAIN}`;
+  private readonly emailFromString = `${process.env.MAILGUN_DEFAULT_NAME} <${process.env.MAILGUN_DEFAULT_EMAIL}>`;
+
+  // Private helper function to send a payload to mailgun API.
+  // Exists to reduce code redundancy.
+  private sendMailPayload(params: Record<string, any>): Promise<AxiosResponse> {
+    return axios.post(
+      `${this.mailgunAPIBaseURL}/messages`,
+      {} /* Mailgun needs the payload to be in params, leave body empty */,
+      {
+        auth: { username: "api", password: process.env.MAILGUN_API_KEY },
+        params
+      }
+    );
+  }
```

Comment:
This function should really be in its own service in which we could add it as an import for this service (dependency injection). We should do this because we can swap out this service for a mock service so we can do the Unit tests without actually sending emails. 

See user-auth-service for more reference. MongoDB is a service and in the unit test I swap it out with the TestDatabase service (that uses an in-memory instance of MongoDB rather than the actual mongo server)